sessions/04-using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic/index.qmd

```diff
@@ -2,3 +2,291 @@
+Try it with a few different probability distributions and parameters, e.g. for the parameters given above and a maximum of 2 weeks (14 days) it would be:
+```{r discretised_gamma}
+gamma_pmf <- discretise(pgamma, 14, shape = 5, rate = 1)
+gamma_pmf
+```
+:::
+
+Next we apply a convolution with the discretised incubation period distribution to the time series of infections to generate a time series of symptom onsets.
+
+```{r convolution}
+## function that takes two inputs to convolve a time series with a delay
+## ts: vector of the time series to convolve
+## delay: the probability mass function of the delay, given as a vector of
+## probabilities, corresponding to discrete indices 0, 1, 2 of the discretised
+## delay distribution
+## example: convolve_with_delay(c(10, 14, 10, 10), c(0.1, 0.6, 0.3))
+convolve_with_delay <- function(ts, delay) {
+  max_delay <- length(delay) - 1 ## subtract one because zero-indexed
+  convolved <- vapply(seq_along(ts), \(i) {
+    ## get vector of infections (14-day window)
+    first_index <- max(1, i - max_delay)
+    ts_segment <- ts[seq(first_index, i)]
+    ## take reverse of pmf and reverse if needed
+    pmf <- rev(delay)[seq_len(i - first_index + 1)]
+    ## convolve with delay distribution
+    ret <- sum(ts_segment * pmf)
+    return(round(ret))
+  }, numeric(1))
+  return(convolved)
+}
+```
+
+::: {.callout-tip}
+## Take 5 minutes
+Try to understand the `convolve_with_delay()` function above.
+Try it with a few different time series and delay distributions.
+How would you create the time series of infections using the discretised gamma distribution created above and saved in `gamma_pmf`?
+:::
+
+::: {.callout-note collapse="true"]}
+```{r applied_convolution}
+onsets <- convolve_with_delay(inf_time_series$n, gamma_pmf)
+```
+:::
+
+We can plot these symptom onsets
+```{r convolution_plot}
+combined <- inf_time_series |>
+  rename(date = infection_date, infections = n) |>
+  mutate(onsets = onsets)
+ggplot(combined, aes(x = date, y = onsets)) +
+  geom_bar(stat = "identity")
+```
+
+Do they look similar to the plot of symptom onsets in the previous session?
+
+## Observation uncertainty
+
+Usually not all data are perfectly observed.
+The counting process itself generally introduces a natural form of error that we can, once again, model with a probability distribution.
+For modelling counts, a common choice is the Poisson distribution.
+We can use this to generate uncertainty around our convolved data.
+
+```{r uncertain}
+onsets <- onsets |>
+  mutate(observed = rpois(n(), onsets))
+```
+
+Does a plot of these observations look more like the data from the previous session than the convolution plotted above?
+
+# Estimating a time series of infection
+
+We now invert the procedure of above, i.e. instead of convolving infects and adding uncertainty to simulate observed symptom onsets, we start with observed simulated onsets to estimate the number of infections, given an incubation period.
```

Comment:
I would really rather we not describe any form of model as backwards process as it seems misleading. 

really we start with somee prior model of infections and work forward based on what we know about the observation process. I think this will make the various add ons to the model seem a lot more logical than backwards steps and also makes it clearer what the dangers are

Reply:
Have rephrased.